### PR TITLE
Added Enterprise Data Exports Report

### PR DESCRIPTION
**Pull request by mjriley**
## Product Description
Enterprise Dashboard:
image

API:
image

Generated Report:
image

## Technical Summary
Associated ticket: https://dimagi.atlassian.net/browse/SAAS-16298

## Feature Flag
No feature flag, as we're doing iterative releases for this

### Safety story
Tested locally

### Automated test coverage

No automated tests so far. The generation logic should eventually be moved into its own functionality. That is the functionality I'd like to write tests for, but I think that is pending the approval of the iterators PR, which sets the precedent for this.

### QA Plan

TBD, but this should go through QA.
QA should verify that the following are listed on the report/API:
- Form Export
- Case Export
- Form Daily Saved Export
- Case Daily Saved Export
- Form Excel Integration
- Case Excel Integration

Nothing else should appear on the report, so its valid to test that something like PowerBI feeds do not appear here.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/enterprise/api/resources.py [hunk 1]
```diff
@@ -388,3 +388,33 @@ def dehydrate(self, bundle):
...
+        bundle.data['domain'] = bundle.obj[0]
+        bundle.data['name'] = bundle.obj[1]
+        bundle.data['export_type'] = bundle.obj[2]
+        bundle.data['export_subtype'] = bundle.obj[3]
+        bundle.data['owner'] = bundle.obj[4]
+
+        return bundle
+
+    def get_primary_keys(self):
+        return ('domain', 'export_type', 'export_subtype', 'name')
```
Comment (jingcheng16) [inline corehq/apps/enterprise/api/resources.py:420]: I tested on staging, I can have two export in same domain, with same export_type, export_subtype and name. In this case, adding owner won't be helpful. Should we add some other attribute like export_id?

File: corehq/apps/enterprise/enterprise.py [hunk 2]
```diff
@@ -383,3 +393,68 @@ def _get_individual_export_rows(self, exports, export_line_counts):
             )
 
         return rows
+
+
+class EnterpriseDataExportReport(EnterpriseReport):
+    title = gettext_lazy('Data Exports')
+
+    def __init__(self, account, couch_user):
```
Comment (jingcheng16) [inline corehq/apps/enterprise/enterprise.py:401]: Tada! Got you! Same issue as mine when you reviewed my PR lol.

File: corehq/apps/enterprise/enterprise.py [hunk 3]
```diff
@@ -383,3 +393,68 @@ def _get_individual_export_rows(self, exports, export_line_counts):
...
+        ]
+
+    def type_lookup(self, doc_type):
+        from corehq.apps.export.models.new import FormExportInstance, CaseExportInstance
+        if doc_type == FormExportInstance.__name__:
+            return _('Form')
+        elif doc_type == CaseExportInstance.__name__:
+            return _('Case')
+        else:
+            return _('Unknown')
```
Comment (jingcheng16) [inline corehq/apps/enterprise/enterprise.py:421]: In what case it will be unknown?

File: corehq/apps/enterprise/enterprise.py [hunk 4]
```diff
@@ -383,3 +393,68 @@ def _get_individual_export_rows(self, exports, export_line_counts):
...
+
+    SUBTYPE_MAP = {
+        is_standard: gettext_lazy('Standard'),
+        is_daily_saved_export: gettext_lazy('Daily Saved Export'),
+        is_excel_integration: gettext_lazy('Excel Integration'),
```
Comment (jingcheng16) [inline corehq/apps/enterprise/enterprise.py:426]: Why we don't support PowerBi/tableau integration?

**Review by jingcheng16 — COMMENTED** [review]
Left a few comment also left a question in tech spec.

File: corehq/apps/enterprise/enterprise.py (hunk 4 above)
Comment (mjriley) [reply corehq/apps/enterprise/enterprise.py:426]: We do! It's just that PowerBI/Tableau Integration is otherwise known as "OData Exports"...and we have a separate report for that.

File: corehq/apps/enterprise/enterprise.py (hunk 3 above)
Comment (mjriley) [reply corehq/apps/enterprise/enterprise.py:421]: This is just defensive programming. The value should never be something else, but if it was and we didn't handle it, then the report would fail to generate entirely and the user would receive no notification -- a bad user experience. If we do run into some odd doc type (this has happened elsewhere, where somehow the type is now something like 'FormExportInstance-Deleted') I think its better to give the report to the user, with a relatively harmless 'Unknown', and allow the user to report it to us if it was something unexpected.

Especially because we have other export types (i.e. `SMSExportInstance`), and all of those types are catalogued by the same Couch index, it isn't out of the realm of possibility that some odd doc type is fetched.

File: corehq/apps/enterprise/api/resources.py (hunk 1 above)
Comment (mjriley) [reply corehq/apps/enterprise/api/resources.py:420]: That's a good catch. I'm not sure its necessary, however, for us to treat these primary keys as typical database primary keys, where conflicts must be avoided. Given that OData feeds are also data exports, I was using `ODataFeedResource` as a comparison point, and that has a primary key of just `report_name`, when we allow multiple odata feeds to share the same name within a domain.  The more important point, to me, is that the OData report has not provided any real way to differentiate between duplicate names (unless you count `report_rows`, but (name, report_rows) isn't guaranteed to be unique either) and, to my knowledge, no one has complained. 

Basically -- making a real primary key would come at the cost of making the report more complex (adding an additional ID field). In practice, I don't believe a customer would ever want to duplicate names. If PowerBI handles primary keys not being unique, I think we are probably fine leaving the definition as is

Comment (mjriley) [reply corehq/apps/enterprise/api/resources.py:420]: Just verified using duplicate report names for the OData Feed, and PowerBI handled it without any complaints. Unless you think we should be including the ID anyway, I think I'd prefer to keep this as is

File: corehq/apps/enterprise/enterprise.py (hunk 2 above)
Comment (mjriley) [reply corehq/apps/enterprise/enterprise.py]: 5eda9c7b00606788a56b704ae4fd78333f6eb397

File: corehq/apps/enterprise/api/resources.py (hunk 1 above)
Comment (jingcheng16) [reply corehq/apps/enterprise/api/resources.py:444]: Primary keys are conventionally unique for good reason—they simplify row identification and avoid ambiguities. I just think it is the best practice. But I see your points too... I don't know. @biyeun thoughts?

Comment (mjriley) [reply corehq/apps/enterprise/api/resources.py:444]: I think the name 'primary key' is misleading. Given my testing, this primary key not being unique does not cause errors, like it would in a typical database. Our objective is to give the users a feed they can import into powerBI, not conform strictly to the OData spec.

**Review by jingcheng16 — APPROVED** [review]
Approved. It's just primary key being non-unique caught me off. Maybe you're right the name "primary key" is misleading. Anyway, I decide to approve it. If something goes wrong in the future, we can still fix it! And we will learn our lesson too.
